api/docker/jobservice.go

```diff
@@ -0,0 +1,108 @@
+package docker
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"strconv"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/api/types/strslice"
+	"github.com/docker/docker/client"
+	"github.com/portainer/portainer"
+	"github.com/portainer/portainer/archive"
+)
+
+// JobService represnts a service that handles jobs on the host
+type JobService struct {
+	DockerClientFactory *ClientFactory
+}
+
+// NewJobService returns a pointer to a new job service
+func NewJobService(dockerClientFactory *ClientFactory) *JobService {
+	return &JobService{
+		DockerClientFactory: dockerClientFactory,
+	}
+}
+
+// Execute will execute a script on the endpoint host with the supplied image as a container
+func (service *JobService) Execute(endpoint *portainer.Endpoint, image string, script []byte) error {
+	buffer, err := archive.TarFileInBuffer(script, "script.sh", 0700)
+	if err != nil {
+		return err
+	}
+
+	cli, err := service.DockerClientFactory.CreateClient(endpoint)
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+
+	pullImage(cli, image)
+
+	// TODO this will run only bash scripts
+	cmd := make([]string, 2)
+	cmd[0] = "sh"
+	cmd[1] = "/tmp/script.sh"
+
+	containerConfig := &container.Config{
+		AttachStdin:  true,
+		AttachStdout: true,
+		AttachStderr: true,
+		Tty:          true,
+		WorkingDir:   "/tmp",
+		Image:        image,
+		Labels: map[string]string{
+			"io.portainer.job.endpoint": strconv.Itoa(int(endpoint.ID)),
+		},
+		Cmd: strslice.StrSlice(cmd),
```

Comment:
Regard comment above we can use a more inline style:

```suggestion
		Cmd: strslice.StrSlice([]string{"sh", "/tmp/script.sh"}),		
```